Plutarch/Internal.hs

```diff
@@ -152,59 +156,26 @@ plam' f = Term $ \i ->
-    getArityBuiltin (RBuiltin PLC.SliceByteString) = Just 2
-    getArityBuiltin (RBuiltin PLC.LengthOfByteString) = Just 0
-    getArityBuiltin (RBuiltin PLC.IndexByteString) = Just 1
-    getArityBuiltin (RBuiltin PLC.EqualsByteString) = Just 1
-    getArityBuiltin (RBuiltin PLC.LessThanByteString) = Just 1
-    getArityBuiltin (RBuiltin PLC.LessThanEqualsByteString) = Just 1
-    getArityBuiltin (RBuiltin PLC.Sha2_256) = Just 0
-    getArityBuiltin (RBuiltin PLC.Sha3_256) = Just 0
-    getArityBuiltin (RBuiltin PLC.Blake2b_256) = Just 0
-    getArityBuiltin (RBuiltin PLC.VerifySignature) = Just 2
-    getArityBuiltin (RBuiltin PLC.AppendString) = Just 1
-    getArityBuiltin (RBuiltin PLC.EqualsString) = Just 1
-    getArityBuiltin (RBuiltin PLC.EncodeUtf8) = Just 0
-    getArityBuiltin (RBuiltin PLC.DecodeUtf8) = Just 0
-    getArityBuiltin (RForce (RBuiltin PLC.IfThenElse)) = Just 2
-    getArityBuiltin (RForce (RBuiltin PLC.ChooseUnit)) = Just 1
-    getArityBuiltin (RForce (RBuiltin PLC.Trace)) = Just 1
-    getArityBuiltin (RForce (RForce (RBuiltin PLC.FstPair))) = Just 0
-    getArityBuiltin (RForce (RForce (RBuiltin PLC.SndPair))) = Just 0
-    getArityBuiltin (RForce (RForce (RBuiltin PLC.ChooseList))) = Just 2
-    getArityBuiltin (RForce (RBuiltin PLC.MkCons)) = Just 1
-    getArityBuiltin (RForce (RBuiltin PLC.HeadList)) = Just 0
-    getArityBuiltin (RForce (RBuiltin PLC.TailList)) = Just 0
-    getArityBuiltin (RForce (RBuiltin PLC.NullList)) = Just 0
-    getArityBuiltin (RForce (RBuiltin PLC.ChooseData)) = Just 5
-    getArityBuiltin (RBuiltin PLC.ConstrData) = Just 1
-    getArityBuiltin (RBuiltin PLC.MapData) = Just 0
-    getArityBuiltin (RBuiltin PLC.ListData) = Just 0
-    getArityBuiltin (RBuiltin PLC.IData) = Just 0
-    getArityBuiltin (RBuiltin PLC.BData) = Just 0
-    getArityBuiltin (RBuiltin PLC.UnConstrData) = Just 0
-    getArityBuiltin (RBuiltin PLC.UnMapData) = Just 0
-    getArityBuiltin (RBuiltin PLC.UnListData) = Just 0
-    getArityBuiltin (RBuiltin PLC.UnIData) = Just 0
-    getArityBuiltin (RBuiltin PLC.UnBData) = Just 0
-    getArityBuiltin (RBuiltin PLC.EqualsData) = Just 1
-    getArityBuiltin (RBuiltin PLC.MkPairData) = Just 1
-    getArityBuiltin (RBuiltin PLC.MkNilData) = Just 0
-    getArityBuiltin (RBuiltin PLC.MkNilPairData) = Just 0
+    getArityBuiltin (RBuiltin builtin) =
+      guard (getPolyVarCount builtin == 0) >> pure (getArityBuiltin' builtin)
+    getArityBuiltin (RForce (RBuiltin builtin)) =
+      guard (getPolyVarCount builtin == 1) >> pure (getArityBuiltin' builtin)
+    getArityBuiltin (RForce (RForce (RBuiltin builtin))) =
+      guard (getPolyVarCount builtin == 2) >> pure (getArityBuiltin' builtin)
     getArityBuiltin _ = Nothing
 
+    getArityBuiltin' :: PLC.DefaultFun -> Natural
+    getArityBuiltin' builtin =
+      case PLC.toBuiltinMeaning @_ @_ @(PLC.Term PLC.TyName PLC.Name _ _ ()) builtin of
+        PLC.BuiltinMeaning sch _ _ -> intToNatural $ PLC.countTermArgs sch
```

Comment:
I just made #72 and with the new tests, and this fails. You have an off-by-one error here. You need to do `-1` to get it to work.

Reply:
Thanks; resolved now.

(Also: https://github.com/Plutonomicon/plutarch/pull/72#discussion_r779764290 )